aws/aws.go

```diff
@@ -1516,20 +1518,38 @@ func extractNodeAddresses(instance *ec2.Instance) ([]v1.NodeAddress, error) {
 	addresses := []v1.NodeAddress{}
 
 	// handle internal network interfaces
+	deviceIPs := make(map[int64][]string)
 	for _, networkInterface := range instance.NetworkInterfaces {
```

Comment:
You should be able to add. If `aws.StringValue(networkInterface.Status) != ec2.NetworkInterfaceStatusInUse` then attachment will be nil and it'll panic, but whatevs.

```
sort.Slice(instance.NetworkInterfaces, func(i, j int) bool {
 return aws.Int64Val(instance.NetworkInterfaces.Attachment.DeviceIndex) < aws.Int64Val(instance.NetworkInterfaces.Attachment.DeviceIndex) 
})
```

and it should "just work"?

Reply:
(That shouldn't happen, unless AWS SDKs somehow returned invalid data, or eventually consistent results.  :) )